plugin/storage/cassandra/factory.go

```diff
@@ -71,10 +73,11 @@ func (f *Factory) InitFromViper(v *viper.Viper) {
 }
 
 // Initialize implements storage.Factory
-func (f *Factory) Initialize(metricsFactory metrics.Factory, logger *zap.Logger) error {
+func (f *Factory) Initialize(metricsFactory metrics.Factory, logger *zap.Logger, reporter healthcheck.StatusReporter) error {
 	f.primaryMetricsFactory = metricsFactory.Namespace(metrics.NSOptions{Name: "cassandra", Tags: nil})
 	f.archiveMetricsFactory = metricsFactory.Namespace(metrics.NSOptions{Name: "cassandra-archive", Tags: nil})
 	f.logger = logger
+	f.StatusReporter = reporter
 
 	primarySession, err := f.primaryConfig.NewSession()
 	if err != nil {
```

Comment:
if Cassandra is not ready, this will cause the application to exit

Reply:
Sorry, I took a while to move back on this.
 
I think my previous explanation was bad.
Actually, the symptom is triggered by cassandra restart. In that case, the session pool become empty after the initialization. In addition, gocql doesn't deal with the reconnection, which makes the jaeger services unavailable after that.